src/main/java/de/hs_mannheim/informatik/ct/persistence/services/DynamicContentService.java

```diff
@@ -73,34 +91,47 @@ public void writeContactList(Collection<Contact<?>> contacts, Visitor target, Co
         }
     }
 
-    private XWPFDocument getRoomsPrintOutDox(List<Room> rooms, Function<Room, UriComponents> uriConverter) throws IOException, XmlException {
-        DocxTemplate.TextTemplate<Room> textReplacer = (room, templatePlaceholder) -> {
-            switch (templatePlaceholder) {
-                case "g":
-                    return room.getBuildingName();
-                case "r":
-                    return room.getName();
-                case "l":
-                    return uriConverter.apply(room).toUriString();
-                case "p":
-                    return Integer.toString(room.getMaxCapacity());
-                case "c":
-                    return room.getRoomPin();
-                default:
-                    throw new UnsupportedOperationException("Template contains invalid placeholder: " + templatePlaceholder);
-            }
-        };
+    public List<XWPFDocument> getRoomsPrintOutDox(List<Room> rooms, Function<Room, UriComponents> uriConverter, boolean privileged) throws IOException, XmlException {
```

Comment:
This has a typo in the method name, should be `getRoomsPrintOutDocx`. But I would change it to only accept and return a single room and document respectively. The benefit of iterating over the rooms outside of this function, is that currently you generate all documents in a building and return them as one huge bundle. If you generate each document add it to the stream immediately, you'll use less memory and the streaming download progress will be much smoother.